runtime-sdk/modules/contracts/src/lib.rs

```diff
@@ -562,6 +574,12 @@ impl<Cfg: Config> Module<Cfg> {
         };
         // Pre-upgrade invocation must succeed for the upgrade to proceed.
         let result = wasm::pre_upgrade::<Cfg, C>(&mut exec_ctx, &contract, &body);
+
+        // Always return success in CheckTx, as we might not have up-to-date state.
```

Comment:
This will skip estimating gas for processing the result and calling the post upgrade handler and result processing.